src/visualization/makie_visualization.jl

```diff
@@ -0,0 +1,373 @@
+  wire_mesh_bottom = Makie.lines!(ax, wire_points, color=:white)
+  Makie.translate!(wire_mesh_top, 0, 0, 1e-3)
+  Makie.translate!(wire_mesh_bottom, 0, 0, -1e-3)
+
+  # draw a flattened mesh wireframe below the solution
+  function compute_z_offset(solution_z)
+      zmin = minimum(solution_z)
+      zrange = (x->x[2]-x[1])(extrema(solution_z))
+      return zmin - .25*zrange
+  end
+  z_offset = Makie.@lift(compute_z_offset($solution_z))
+  get_flat_points(wire_points, z_offset) = [Makie.Point(point.data[1:2]..., z_offset) for point in wire_points]
+  flat_wire_points = Makie.@lift get_flat_points($wire_points, $z_offset)
+  wire_mesh_flat = Makie.lines!(ax, flat_wire_points, color=:black)
+
+  # reset colorbar each time solution changes
+  Makie.Colorbar(fig[1, 3], limits = Makie.@lift(extrema($solution_z)), colormap = :viridis, flipaxis = false)
+
+  # syncs the toggle to the mesh
+  Makie.connect!(wire_mesh_top.visible, toggle_solution_mesh.active)
+  Makie.connect!(wire_mesh_bottom.visible, toggle_solution_mesh.active)
+  Makie.connect!(wire_mesh_flat.visible, toggle_mesh.active)
+
+  # On OSX, shift-command-4 for screenshots triggers a constant "up-zoom".
+  # To avoid this, we remap up-zoom to the right shift button instead.
+  Makie.cameracontrols(ax.scene).attributes[:up_key][] = Makie.Keyboard.right_shift
+
+  # typing this pulls up the figure (similar to display(plot!()) in Plots.jl)
+  fig
+end
+
+# ================== new Makie plot recipes ====================
+
+@Makie.recipe(TrixiHeatmap, plot_data_series) do scene
+  Makie.Theme(;)
+end
+
+function Makie.plot!(myplot::TrixiHeatmap)
+  pds = myplot[:plot_data_series][]
+
+  plotting_mesh = generate_plotting_triangulation(pds; set_z_coordinate_zero = true)
+
+  @unpack variable_id = pds
+  pd = pds.plot_data
+  solution_z = vec(StructArrays.component(pd.u, variable_id))
+  Makie.mesh!(myplot, plotting_mesh, color=solution_z, shading=false)
+
+  # Makie hides keyword arguments within `myplot`; see also
+  # https://github.com/JuliaPlots/Makie.jl/issues/837#issuecomment-845985070
+  plot_mesh = if haskey(myplot, :plot_mesh)
+    myplot.plot_mesh[]
+  else
+    true # default to plotting the mesh
+  end
+
+  if plot_mesh
+    xyz_wireframe = generate_plotting_wireframe(pds; set_z_coordinate_zero = true)
+    Makie.lines!(myplot, xyz_wireframe, color=:lightgrey)
+  end
+
+  myplot
+end
+
+# redirects Makie.plot(pd::PlotDataSeries2D) to custom recipe TrixiHeatmap(pd)
+Makie.plottype(::Trixi.PlotDataSeries2D{<:Trixi.UnstructuredPlotData2D}) = TrixiHeatmap
+
+# Makie does not yet support layouts in its plot recipes, so we overload `Makie.plot` directly.
+function Makie.plot(sol::TrixiODESolution;
+                    plot_mesh=true, solution_variables=nothing)
+  return Makie.plot(PlotData2D(sol, solution_variables=solution_variables); plot_mesh=plot_mesh)
+end
+
+# convenience struct for editing Makie plots after they're created.
+struct FigAxes{Axes}
```

Comment:
I'd call this `FigureAndAxes` to make it more descriptive and to avoid the misunderstanding "axes of the figure" (that's how I understood `FixAxes` when I first sawy it). Similarly, use `figure_and_axes` for the variables. The long name shouldn't be an issue since it's just used internally for convenience. OTOH, it's not a big issue for the same reason, so if you do not like this suggestion, it's ok to leave it as it is.